imap_processing/quality_flags.py

```diff
@@ -0,0 +1,48 @@
+"""Bitwise flagging."""
+
+from enum import IntFlag
+
+
+# TODO: as needed create a message for each name, value pair
+#  for a more verbose description of what the flag indicates
+class ImapQualityFlag(IntFlag):
+    """Base Quality flags."""
+
+    @property
+    def name(self) -> str:
+        """
+        Override the default name property to handle combined flags.
+
+        Returns
+        -------
+        combined_name : str
+            The combined name of the individual flags.
+        """
+        if self._name_ is not None:
+            return self._name_
+
+        members = [member for member in BaseQualityFlags if member & self == member]
+        return "|".join(str(m.name) for m in members if m != BaseQualityFlags.NONE)
+
+
+class BaseQualityFlags(ImapQualityFlag):
+    """Base Quality flags."""
+
+    NONE = 0x0
```

Comment:
Do we need "NONE", or can this be implicit? No strong preference either way, so fine to leave as well.